src/actions/filterActions.js

```diff
@@ -14,6 +15,21 @@ export const changeEventFilter = (history) => (filter) => {
   }
 }
 
+export const changeUncorrelatedEventFilter = (history) => (filter) => {
+  filterService.getUrlFromUncorrelatedFilter(history, filter)
+  return {
+    type: CHANGE_EVENT_FILTER,
+    filter
+  }
+}
+
+export const clearFilterIncidentId = (filter) => {
```

Comment:
This doesn't actually clear the Filter Incident Id (though I see you do so manually before calling this function).

The fact that this (an action that specifically controls the incidentId value of the filter) is needed makes me think that the filter reducer would be better off broken up into multiple separate reducers with a combineReducers statement; that would give us more granular control of what gets updated by each action type.

If you'd rather not go that far, you can replace the call to this action in uncorrelatedEvents.js with dispatch(filterActions.synchronizeFilters(filter, null, null, history)), though you'd also need a WithRouter() statement to pull in history.

If you definitely want to keep this action, change the type to something that matches the function name (CLEAR_EVENT_FILTER_INCIDENTID or similar) and add a reducer case for it:

const { incidentId, ...newState } = state
return newState